From ae8a9597d6841d4be8dd29a9c19362b02db9992d Mon Sep 17 00:00:00 2001
From: James Gaboardi <jgaboardi@gmail.com>
Date: Tue, 24 Oct 2023 12:18:19 -0400
Subject: [PATCH] adapt import to libpysal update

Fix missing imports. Taken from a fix in version 1.4.2 at https://github.com/pysal/spreg/commits/v1.4.2
in this commit: https://github.com/pysal/spreg/commit/ae8a9597d6841d4be8dd29a9c19362b02db9992d

---
 spreg/diagnostics.py      | 34 ++++++++++++++++++----------------
 spreg/diagnostics_tsls.py | 16 +++++++---------
 2 files changed, 25 insertions(+), 25 deletions(-)

diff --git a/spreg/diagnostics.py b/spreg/diagnostics.py
index a2fd9e5a..b67da250 100644
--- a/spreg/diagnostics.py
+++ b/spreg/diagnostics.py
@@ -6,9 +6,14 @@
     "Luc Anselin luc.anselin@asu.edu, Nicholas Malizia nicholas.malizia@asu.edu "
 )
 
-from libpysal.common import *
-import scipy.sparse as SP
 from math import sqrt, pi
+
+from libpysal.common import MISSINGVALUE
+import numpy as np
+import numpy.linalg as la
+import scipy.sparse as SP
+from scipy import stats
+
 from .utils import spmultiply, sphstack, spmin, spmax
 
 
diff --git a/spreg/diagnostics_tsls.py b/spreg/diagnostics_tsls.py
index ab1d0a25..5d918747 100644
--- a/spreg/diagnostics_tsls.py
+++ b/spreg/diagnostics_tsls.py
@@ -7,7 +7,8 @@
     "Luc Anselin luc.anselin@asu.edu, Nicholas Malizia nicholas.malizia@asu.edu "
 )
 
-from libpysal.common import *
+import numpy as np
+from scipy import stats
 from scipy.stats import pearsonr
 
 __all__ = ["t_stat", "pr2_aspatial", "pr2_spatial"]
